chia/wallet/wallet_protocol.py

```diff
@@ -28,7 +29,9 @@ def type(cls) -> WalletType:
     def id(self) -> uint32:
         ...
 
-    async def coin_added(self, coin: Coin, height: uint32, peer: WSChiaConnection) -> None:
+    async def coin_added(
+        self, coin: Coin, height: uint32, peer: WSChiaConnection, coin_data: Optional[Streamable]
```

Comment:
Yes, the idea is to reuse the data we got in the WSM. I split PRs, so you will not see changes for specific wallets in this PR. So far, we know we can reuse it in Clawback/CAT/NFT/DID/VC. Some wallets may still need to fetch data on the fly in some cases because the coin_added is not called after the determine_coin_type().

Reply:
The `coin_data` parameter will be of a specific type for each class that implements `WalletProtocol`?  If so, then i think that making the protocol generic would fit.  I think that 'fake implementing' the protocol, sorta, would deserve a `# TODO:` at least at each site that still needs updated, or at least reviewed.  Rather than just an unused parameter which leaves it unclear to readers whether it is intentionally unused or not.

Reply:
For generic, do you mean something like T = TypeVar("T") or Streamable is okay?

Reply:
`Streamable` expresses nearly nothing of interest around the type you intend here.  https://docs.python.org/3.11/library/typing.html#user-defined-generic-types We have other `Generic` protocols as well.